web/src/settings_components.ts

```diff
@@ -893,6 +894,72 @@ export function check_property_changed(
     return current_val !== proposed_val;
 }
 
+export function populate_data_for_request(
+    subsection: JQuery,
+    for_realm_default_settings: boolean,
+    sub: StreamSubscription | undefined,
+    group: UserGroup | undefined,
+    custom_profile_field?: CustomProfileField | undefined,
+): Record<string, string | boolean | number> {
+    let data: Record<string, string | boolean | number> = {};
+    const properties_elements = get_subsection_property_elements(subsection);
+    for (const input_elem of properties_elements) {
+        const $input_elem = $(input_elem);
+        if (
+            check_property_changed(
+                input_elem,
+                for_realm_default_settings,
+                sub,
+                group,
+                custom_profile_field,
+            )
+        ) {
+            const input_value = get_input_element_value(input_elem);
+            if (input_value !== undefined) {
+                let property_name: string;
+                if (
+                    for_realm_default_settings ||
+                    sub !== undefined ||
+                    group !== undefined ||
+                    custom_profile_field !== undefined
+                ) {
+                    property_name = extract_property_name($input_elem, for_realm_default_settings);
+                } else if ($input_elem.attr("id")!.startsWith("id_authmethod")) {
+                    // Authentication Method component IDs include authentication method name
+                    // for uniqueness, anchored to "id_authmethod" prefix, e.g. "id_authmethodapple_<property_name>".
+                    // We need to strip that whole construct down to extract the actual property name.
+                    // The [\da-z]+ part of the regexp covers the auth method name itself.
+                    // We assume it's not an empty string and can contain only digits and lowercase ASCII letters,
+                    // this is ensured by a respective allowlist-based filter in populate_auth_methods().
+                    const match_array = /^id_authmethod[\da-z]+_(.*)$/.exec(
+                        $input_elem.attr("id")!,
+                    );
+                    assert(match_array !== null);
+                    property_name = match_array[1];
+                } else {
+                    const match_array = /^id_realm_(.*)$/.exec($input_elem.attr("id")!);
+                    assert(match_array !== null);
+                    property_name = match_array[1];
+                }
+
+                if (property_name === "stream_privacy") {
+                    data = {
+                        ...data,
+                        ...settings_data.get_request_data_for_stream_privacy(
+                            input_value!.toString(),
+                        ),
+                    };
+                    continue;
+                }
+                assert(input_value !== null);
```

Comment:
Can we remove this `assert` and check `input_value !== null` directly in the if-condition? In what cases `input_value` will be `null` here?